### PR TITLE
Android start on boot (auto start after reboot)

### PR DESCRIPTION
**Pull request by Heap-Hop**
https://github.com/rustdesk/rustdesk/discussions/1516

- [x] Enable BootReceiver.
- [x] Adapt to Android 12 (ACTION_IGNORE_BATTERY_OPTIMIZATION_SETTINGS, SYSTEM_ALERT_WINDOW).
- [x] add `Start on Boot` setting option.
- [x] Permissions detection before `Start on Boot` is enabled.
- [x] Add start_service and pass app_dir from Android MainService.
- [x] Added UI `Start on Boot`

- Optimized and refactored some Android code.

### CODE REVIEW

File: src/lang/pl.rs [hunk 1]
```diff
@@ -456,9 +458,6 @@ pub static ref T: std::collections::HashMap<&'static str, &'static str> =
         ("Reconnect", "Połącz ponownie"),
         ("Codec", "Kodek"),
         ("Resolution", "Rozdzielczość"),
-        ("Use temporary password", "Użyj hasła tymczasowego"),
-        ("Set temporary password length", "Ustaw długość hasła tymczasowego"),
```
Comment (Orholt) [inline src/lang/pl.rs:460]: Why did you remove those lines?

Comment (Heap-Hop) [reply src/lang/pl.rs:460]: Not my intention, just ran our res/lang.py.

Comment (rustdesk) [reply src/lang/pl.rs:460]: "Use temporary password" is still be used.

Comment (rustdesk) [reply src/lang/pl.rs:460]: “"Use temporary password" is changed to ""Use one-time password".

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/3322

**Comment by rustdesk** [comment]
#3275

**Comment by rustdesk** [comment]
#2529

**Comment by rustdesk** [comment]
#2034

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/3533#issuecomment-1490065448

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/3709

**Comment by marcatmadu** [comment]
>Added UI Start on Boot
is this acomplished in the Play Store app ?   I installed rustdesk on my android but I cannot find an UI option called "Start on Boot" ?!?

**Comment by michalwitek1232** [comment]
> '>Added UI Start on Boot
> 
> is this acomplished in the Play Store app ? I installed rustdesk on my android but I cannot find an UI option called "Start on Boot" ?!?

Which version did you downloaded? I guess it's old stable on Play Store, Install rustdesk nightly from proper apk file

**Comment by marcatmadu** [comment]
@michalwitek1232   you were right, I was using the old play store version. Now I installed the latest nightly version (1.1.10-1) and still see no option named "Start on Boot".   Also when service is running, after reboot of the phone, service is not running again.

**Comment by liangjaden** [comment]
I installed the latest nightly version (1.1.10-1) and still see no option named "Start on Boot". me too.

**Comment by michalwitek1232** [comment]
Weird, I mean this option is here 
Screenshot_2023-06-25-19-50-59-932-edit_com carriez flutter_hbb
(Should be called "Autostart", "Startup" or something like this)

**Comment by liangjaden** [comment]
@michalwitek1232 Thank you, I download nightly version(1.2.0) from github Releases , Now it's work well~

**Comment by marcatmadu** [comment]
can someone post a link to this version 1.2 ?  When I go to nightly releases, version 1.1.9 is the latest ....

**Comment by liangjaden** [comment]
@marcatmadu https://github.com/rustdesk/rustdesk/releases/tag/nightly

**Comment by marcatmadu** [comment]
got it now but it's not working on Realme 8 Pro. (Android 12)

**Comment by michalwitek1232** [comment]
> got it now but it's not working on Realme 8 Pro. (Android 12)

I mean I also have a problem with this feature but on POCO F2 Pro. Maybe it's root issue idk.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/discussions/1330#discussioncomment-6860073
